spotify-artist-viewer/Models/SARequestManager.m

```diff
@@ -0,0 +1,118 @@
+    //GET https://api.spotify.com/v1/search
+    //create url based on query
+    NSString *urltest = @"https://api.spotify.com/v1/search?q=";
+    NSString *url = [urltest stringByAppendingString:query];
+    NSString *type = @"&type=artist";
+    NSString *urlWithType = [url stringByAppendingString:type];
+    NSURL *requestURL = [NSURL URLWithString:urlWithType];
+       
+    //create request, set to GET
+    NSMutableURLRequest *urlRequest = [[NSMutableURLRequest alloc] initWithURL:requestURL];
+    [urlRequest setHTTPMethod:@"GET"];
+    
+    NSURLSessionConfiguration *configuration = [NSURLSessionConfiguration defaultSessionConfiguration];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration];
+   
+   
+    NSMutableArray *artists = [[NSMutableArray alloc] init];
+    NSURLSessionDataTask *task = [session dataTaskWithRequest:urlRequest completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+      NSDictionary *result = [NSJSONSerialization JSONObjectWithData:data
+                                                  options:0
+                                                    error:NULL];
+ 
+        NSLog(@"JSON - %@", result);
+        if([result isKindOfClass:[NSDictionary class]]){
+           NSDictionary *itemArray = [[NSDictionary alloc] init];
+            itemArray = [result objectForKey:@"artists"];
+    
+         
+           
+            NSDictionary *artistArray = [[NSDictionary alloc] init];
+            artistArray = [itemArray objectForKey:@"items"];
+
+            
+            for(itemArray in [itemArray valueForKey:@"items"]){
+                SAArtist *a = [[SAArtist alloc]init];
+                
+                NSDictionary *imgArray = [[NSDictionary alloc] init];
+                imgArray = [itemArray objectForKey:@"images"];
+                NSArray *tempimgUrl = [imgArray valueForKey:@"url"];
+                
+                //making sure artist has images
+                
+                if(tempimgUrl.count != NULL){
+                
+                a.imgURL = [tempimgUrl objectAtIndex:0];
+                }
+                
+                NSLog(@"url - %@", tempimgUrl);
+                NSString *name = [itemArray valueForKey:@"name"];
+                                a.artistName = name;
+                
+                NSString *uri = [itemArray valueForKey:@"uri"];
+                a.artistUri = uri;
+                
+                NSString *regQuery = @"http://developer.echonest.com/api/v4/artist/biographies?api_key=FILDTEOIK2HBORODV&id=";
+            
+                
+                                              
+                [artists addObject:a];
+                               
+
+            }
+            
+            
+            //econest query http://developer.echonest.com/api/v4/artist/biographies?api_key=FILDTEOIK2HBORODV&id=spotify:artist:4Z8W4fKeB5YxbusRsdQVPb
+            
+            
+            success(artists);
```

Comment:
Wrap the success in an if (success) to make sure that the success block in not nil.